File: Summer 2024 Health Monitoring - Charlie's Work/Charlie_Code/Data Analysis and Matching/Python/HR_BP_Filter.py
import numpy as np
from scipy.signal import butter, filtfilt

def Baseline_Removal(sig, fs):
    order = 8   # order of filter
    fcutlow = 0.83   # low cut frequency in Hz
    fcuthigh = 1.7   # high cut frequency in Hz
    nyquist = 0.5 * fs
    low = fcutlow / nyquist
    high = fcuthigh / nyquist
    b, a = butter(order, [low, high], btype='band')
    filt_sig = filtfilt(b, a, sig)
    return filt_sig

File: Summer 2024 Health Monitoring - Charlie's Work/Charlie_Code/Data Analysis and Matching/MATLAB/HR_BP_Filter.m
function filt_sig = Baseline_Removal(sig,fs)
% High-pass filter
order=8;   %order of filter
fcutlow = 0.83;   %low cut frequency in Hz
fcuthigh = 1.7;   %high cut frequency in Hz
[b,a]=butter(order,[fcutlow,fcuthigh]/(fs/2),'bandpass');   % fs/2 term in the cutoff freq is the Nyquist Sampling rate 
filt_sig = filter(b,a,sig);  %filtered signal
end